### PR TITLE
Code fixes

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

A mass pull request that fixes as many bugs found by pylint as possible, in order to fix CI.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/message.py [hunk 1]
```diff
@@ -1152,42 +1159,42 @@ async def delete(delay: float):
...
-        view: Optional[View] = ...,
+            self,
+            *,
+            content: Optional[str] = ...,
+            embed: Optional[Embed] = ...,
+            attachments: List[Attachment] = ...,
+            suppress: bool = ...,
+            delete_after: Optional[float] = ...,
+            allowed_mentions: Optional[AllowedMentions] = ...,
+            view: Optional[View] = ...,
```
Comment (Dorukyum) [inline discord/message.py:1170]: Why were these indented? Shouldn't they stay at 4 spaces?

File: discord/message.py [hunk 2]
```diff
@@ -998,7 +998,7 @@ def is_system(self) -> bool:
         )
 
     @utils.cached_slot_property('_cs_system_content')
-    def system_content(self):
+    def system_content(self) -> str:
```
Comment (zeffo) [inline discord/message.py:1001]: There has to be a better way to do this than chained ifs no?

File: discord/message.py (hunk 1 above)
Comment (zeffo) [reply discord/message.py:1170]: pylint sucks

Comment (Lulalaby) [reply discord/message.py:1170]: lol

Comment (BobDotCom) [reply discord/message.py:1170]: Actually, they were indented to comply with PEP8. When parameters are formatted like this, they should be further indented to distinguish from the code inside the following block.

File: discord/message.py (hunk 2 above)
Comment (BobDotCom) [reply discord/message.py:1001]: Match-case crossed my mind but that's not an option for now, at least.

**Comment by BobDotCom** [comment]
?

**Comment by Lulalaby** [comment]
I think we should merge all typing / lint prs into one and merge it after a big test into master.

**Comment by BobDotCom** [comment]
Bumped to 2.1 to prevent feature creep

**Comment by Lulalaby** [comment]
Please re-do pr to master

**Pull request closed without merging**